src/t_set.c

```diff
@@ -51,6 +51,16 @@ robj *setTypeCreate(sds value, size_t size_hint) {
     return createSetObject();
 }
 
+/* Check if the existing set should be converted to another encoding based off the
+ * the size hint. */
+void setTypeTryConvert(robj *set, size_t size_hint) {
+    if ((set->encoding == OBJ_ENCODING_LISTPACK && size_hint >= server.set_max_listpack_entries)
+        || (set->encoding == OBJ_ENCODING_INTSET && size_hint >= server.set_max_intset_entries))
+    {
+        setTypeConvert(set, OBJ_ENCODING_HT);
```

Comment:
There's a function to dimension the hashtable so it doesn't need rehashing later.

```suggestion
        setTypeConvertAndExpand(set, OBJ_ENCODING_HT, size_hint, 1);
```

Reply:
I originally avoided this since I didn't want to oversize the hash in case there were duplicates.

Reply:
we should do the same in setTypeCreate (use the size_hint to expand).
or move the call to setTypeMaybeConvert outside the `else`.

maybe we also wanna do it in hashTypeTryConversion? (at least in the case where we call hashTypeConvert)

Reply:
Sounds good. I'll update hashTypeConvert as well.

Reply:
We decided to use a dict instead of a listpack just because we assumed the args don't contain any duplicates, so it's basically the same reasoning here, i.e. we still assume the args don't contain any duplicates, right?

Reply:
yeah, it'll work correctly, but is optimized for the case people don't provide duplicates in one command.

Reply:
Yes, the concern I have is that someone using duplicates may start using more memory when they pickup this version, although it will be faster.